app/Models/ComponentGroup.php

```diff
@@ -141,4 +162,30 @@ public function getPresenterClass()
     {
         return ComponentGroupPresenter::class;
     }
+
+    /**
+     * Finds all component groups which are visible to public.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     *
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeVisible(Builder $query)
```

Comment:
Might be worth adding the inverse of this, especially with Plugins on the horizon. Thoughts?